### ISSUE TITLE
Problem with symlinks over sshfs

### ISSUE THREAD

**Issue by tardigradus** [magit/magit#2213]
I have a symlink which magit seem to think has been deleted and recreated (status 'Deleted' or 'Typechange').  However, the link has not in fact been changed. 

Attempting to revert the change with 'v' gives the following error:

255 git --no-pager -c core.preloadindex=true checkout -- foo/bar.cfg
error: unable to create symlink foo/bar.cfg (Input/output error)

The symlink in question is on a remote server accessed via sshfs.

**Comment by tarsius** [comment]
Please try to come up with reproducible steps. Also try to reproduce without sshfs.

**Comment by tardigradus** [comment]
OK, I know what the problem is:  I'm using git 2.1.4 on my local machine, but using git 1.7.1 on the server.

I created a git repo on the server with a single file and a soft link to it.  I then accessed it via sshfs.
Commiting via magit causes git on the command line to see a typechange. If I commit this change on the command line, magit then sees an unstaged typechange. This seems to continue:

$ git log --oneline
683a030 Another typchange seen by magit
8c987a7 Spontaneous typechange II
f8f0c53 Spontaneous typechange seen by magit
f24aeea Spontaneous typechange
f1c1e29 Initial commit

I the *magit-process* buffer I see

 1 file changed, 1 insertion(+), 1 deletion(-)
 rewrite link (100%)
 mode change 120000 => 100644

Although I am not using git explicity on the server, I do have the status of the repository in my bash prompt.  This calls the server's version of git, which seems to treat symlinks differently to the way my local version does.

TL;DR

If you use magit with sshfs, make sure you don't (inadvertantly) run command from a different git version on the remote machine.

**Comment by tarsius** [comment]
Glad you could figure out what happened.
